### PR TITLE
Allow publishing PR branches to npm for testing

### PR DESCRIPTION
**Pull request by SomeHats**
If you add the `publish-packages` label to a PR, we'll publish it to npm under the `internal` release tag.

### Change type

- [x] `other`

### CODE REVIEW

File: scripts/publish-prerelease.ts [hunk 1]
```diff
@@ -22,7 +23,15 @@ async function main() {
...
 
-main()
+const args = process.argv.slice(2)
+if (args.length === 0) {
+	throw new Error('Usage: publish-prerelease.ts <releaseTag>')
+}
+const releaseTag = args[0]
+if (releaseTag !== 'canary' && releaseTag !== 'latest') {
+	throw new Error('releaseTag must be "canary" or "latest"')
+}
```
Comment (ds300) [inline scripts/publish-prerelease.ts:36]: won't this fail when you invoke it with `internal`?

File: .github/workflows/publish-branch.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    environment: npm deploy
+    timeout-minutes: 60
+    runs-on: ubuntu-latest-16-cores-open
+    if: github.event.label.name == 'publish-packages'
+
+    permissions:
+      pull-requests: write
+
+    steps:
+      - name: Remove the update-snapshots label
```
Comment (ds300) [inline .github/workflows/publish-branch.yml:23]:
```suggestion
      - name: Remove the publish-packages label
```

**Review by ds300 — APPROVED** [review]
nice!

File: scripts/publish-prerelease.ts (hunk 1 above)
Comment (SomeHats) [reply scripts/publish-prerelease.ts:36]: yes, oops! i blame the robbot. thanks for pointing this out!

**Pull request merged**